src/components/FreestarAdSlot/index.js

```diff
@@ -1,97 +1,71 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import { Promise } from 'q'
-
-const getFreestar = () => {
-  return new Promise((resolve, reject) => {
-    const maxTries = 10
-    let retryCount = 0
-    const waitForFreestarReady = setInterval(() => {
-      if (window.freestar && window.googletag && window.googletag.apiReady) {
-        clearInterval(waitForFreestarReady)
-        resolve(window.freestar)
-      } else if (retryCount === maxTries) {
-        clearInterval(waitForFreestarReady)
-        reject(`freestar NOT ready after ${maxTries} tries`)
-      } else {
-        retryCount++
-      }
-    }, 10)
-  })
-}
 
 class FreestarAdSlot extends Component {
   componentDidMount () {
-    this.newAdSlots()
-  }
-
-  componentWillUnmount () {
-    const { adUnit, onDeleteAdSlotsHook } = this.props
-    getFreestar().then(freestar => {
-      if (this.adSlotIsReady(adUnit)) {
-        freestar.deleteAdSlots(adUnit)
-        onDeleteAdSlotsHook(adUnit.placementName)
-      }
-    })
-  }
+    const { publisher, debug } = this.props
```

Comment:
Pardon my n00bness with React, but somewhere in the chain ?fsdebug=99 will end up as a debug prop here?

Reply:
this was meant to replace this line in the traditional header script:

`freestar.debug = window.location.search.indexOf('fsdebug') === -1 ? false : true;`

after rethinking, there's no need to move it to a prop, i'll update to pull the value from the query string

Reply:
to help with our debugging of a react app i think it would still be helpful to grab things from the url as opposed to passing in a component param since that would require the pub to change code to toggle into QA or not .   Can we add our normal logic decide if we are loading QA or not ( using URL) and then add support for looking at presence of our FS query params to append to the url as well ? thoughts @aecook